### PR TITLE
Correct line and column number in source position when raising LexingError

### PR DESCRIPTION
**Pull request by michal-pawlik**
When lexer is unable to find matching token, `LexingError` raises. This error contains source position for further debugging, but in current implementation it only points valid index (character number), while the line and colum numbers equal `-1`.

This PR fixes this issue by tracking column number in `next()` method of `LexerStream` and providing both `self._lineno` and `self._colno` to `SourcePosition` in `LexingError`.

### CODE REVIEW

File: tests/test_lexer.py [hunk 1]
```diff
@@ -150,3 +150,30 @@ def test_error(self):
...
         assert 'SourcePosition(' in repr(excinfo.value)
+
+    def test_error_line_number(self):
+        lg = LexerGenerator()
+        lg.add("NEW_LINE", r"\n")
+        l = lg.build()
+
+        stream = l.lex("\nfail")
+        with raises(LexingError) as excinfo:
+            stream.next()
```
Comment (alex) [inline tests/test_lexer.py]: Please move this line outside the raises block so it's clear which line raises an exception.

File: tests/test_lexer.py [hunk 2]
```diff
@@ -150,3 +150,30 @@ def test_error(self):
...
+        assert excinfo.value.source_pos.lineno == 2
+
+    def test_error_column_number(self):
+        lg = LexerGenerator()
+        lg.add("NUMBER", r"\d+")
+        lg.add("PLUS", r"\+")
+        l = lg.build()
+        stream = l.lex("1+2+fail")
+        with raises(LexingError) as excinfo:
+            stream.next()
```
Comment (alex) [inline tests/test_lexer.py]: Same here.

**Comment by nobodxbodon** [comment]
Seems there's edge case not covered here:
```python
    def test_error_reset_column_number(self):
        lg = LexerGenerator()
        lg.add("NEW_LINE", r"\n")
        lg.add("NUMBER", r"\d+")
        l = lg.build()
        stream = l.lex("12\nfail")
        stream.next()
        stream.next()
        with raises(LexingError) as excinfo:
            stream.next()

        assert excinfo.value.source_pos.colno == 1   # it fails with 3 instead
```

The same assertion works if added in test_error_line_number.

**Comment by nobodxbodon** [comment]
FYI here's a fix for colno when there's lexing error: https://github.com/nobodxbodon/rply/commit/4afbd7f12f200790d0466c23f1e0660b770e5ef6 , covering the edge case above.
